test/e2e/tests/test_hpo.py

```diff
@@ -61,26 +77,99 @@ def get_sagemaker_hpo_job(sagemaker_client, hpo_job_name):
         )
         return None
 
+def get_hpo_sagemaker_status(hpo_job_name: str):
+    hpo_sm_desc = get_sagemaker_hpo_job(hpo_job_name)
+    return hpo_sm_desc["HyperParameterTuningJobStatus"]
+
+def get_hpo_resource_status(reference: k8s.CustomResourceReference):
+    resource = k8s.get_resource(reference)
+    assert "hyperParameterTuningJobStatus" in resource["status"]
+    return resource["status"]["hyperParameterTuningJobStatus"]
 
 @service_marker
 @pytest.mark.canary
 class TestHPO:
-    def test_create_hpo(self, sagemaker_client, xgboost_hpojob):
+    list_status_created = ("InProgress", "Completed")
+    list_status_stopped = ("Stopped", "Stopping")
+    status_inprogress: str = "InProgress"
+    status_completed: str = "Completed"
+
+    def _wait_resource_hpo_status(
+        self,
+        reference: k8s.CustomResourceReference,
+        expected_status: str,
+        wait_periods: int = 30,
+    ):
+        return wait_for_status(
+        expected_status, wait_periods, get_hpo_resource_status, reference
+    )
+
+    def _wait_sagemaker_hpo_status(
+        self, 
+        hpo_job_name,
+        expected_status: str,
+        wait_periods: int = 30,
+    ):
+        return wait_for_status(
+            expected_status,
+            wait_periods,
+            get_hpo_sagemaker_status,
+            hpo_job_name,
+        )
+
+    def _assert_hpo_status_in_sync(
+        self, hpo_job_name, reference, expected_status
+    ):
+        assert (
+            self._wait_sagemaker_hpo_status(
+                hpo_job_name, expected_status
+            )
+            == self._wait_resource_hpo_status(reference, expected_status)
+            == expected_status
+        )
+
+    def test_hpo(self, xgboost_hpojob):
+        (reference, resource) = xgboost_hpojob
+        assert k8s.get_resource_exists(reference)
+    
+        hpo_job_name = resource["spec"].get("hyperParameterTuningJobName", None)
+        assert hpo_job_name is not None
+
+        hpo_sm_desc = get_sagemaker_hpo_job(hpo_job_name)
+        assert k8s.get_resource_arn(resource) == hpo_sm_desc["HyperParameterTuningJobArn"]
+        assert hpo_sm_desc["HyperParameterTuningJobStatus"] in self.list_status_created
+        assert k8s.wait_on_condition(reference, "ACK.ResourceSynced", "False")
+
+        self._assert_hpo_status_in_sync(
+            hpo_job_name, reference, self.status_inprogress
+        )
+
+        # Delete the k8s resource.
+        _, deleted = k8s.delete_custom_resource(reference)
+        assert deleted is True
+
+        hpo_sm_desc = get_sagemaker_hpo_job(hpo_job_name)
+        assert hpo_sm_desc["HyperParameterTuningJobStatus"] in self.list_status_stopped
+
+    def test_completed_hpo(self, xgboost_hpojob):
```

Comment:
Any reason you are creating another test method (and another HPO Job) instead of keeping the checks on status in the original test method that tests the full "happy path" lifecycle of an HPO Job?

Reply:
Yes, this is essential. 
1. The first test starts the hpoJob -> checks that it gets created -> then stops -> checks that the status in is stopping/stopped in SM and the resource gets deleted from k8s. 
2. The second (new) test starts the hpoJob -> checks that it gets created -> then waits for it to go to completion in SageMaker making sure all the fields work as expected -> it also checks that the k8s status gets updated as expected, ensuring requeues (testing custom code) -> and finally tests that a job completed in SageMaker, can be deleted from k8s (a bug we caught recently).

This second case has already caught a couple of bugs - although just incorrect specs - but then those specs will become our samples so we do need to test both scenarios.

Reply:
also sdkDelete in 
1. calls SageMaker Stop*Job() 
2. retuns early without calling Stop*Job() - again testing custom code
having separate tests ensure both cases work

Reply:
lets just call it `test_completed` and `test_stopped` ? because the its already under TestHPO class